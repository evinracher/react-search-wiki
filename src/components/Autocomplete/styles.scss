.input-suggestions {
  border: 1px solid #dfe1e5;
  border-radius: 10px;
  margin-top: 10px;
  width: 100%;
  right: 0;
  left: 0;
  box-sizing: border-box;

  &-item {
    padding: 0.2rem 1rem;
    text-align: left;
    box-sizing: border-box;
  }

  &-link {
      color: black;
      text-decoration: none;
  }

  .search-link {
    display: block;
    margin-top: 10px;
    padding: 4px 0;
  }
}
