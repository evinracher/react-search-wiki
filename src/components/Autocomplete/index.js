import React from 'react';
import ReactAutocomplete from 'react-autocomplete';
import Input from '../Input';
import './styles.scss';

const Autocomplete = ({ articles, searchValue, onSearchChange }) => {
  return (
    <ReactAutocomplete
      items={articles}
      renderInput={Input}
      inputProps={{ placeholder: 'Input a search term' }}
      // shouldItemRender={(item, value) => item.label.toLowerCase().indexOf(value.toLowerCase()) > -1}
      getItemValue={item => item.label}
      renderMenu={(children, value, style) => {
        return articles && articles.length
          ? (
            <div style={{ ...style }} className="input-suggestions">
              {children}
              <a href={`/search?query=${value}`} className="search-link">
                See all results
              </a>
            </div>
          )
          : <></>;// Empty element
      }
      }
      renderItem={(item, highlighted) =>
        <div
          className="input-suggestions-item"
          key={item.id}
          style={{ backgroundColor: highlighted ? '#eee' : 'transparent' }}
        >
          <a className="input-suggestions-link" href={item.id} target="_blank">{item.label}</a>
        </div>
      }
      value={searchValue}
      onChange={onSearchChange}
    />
  );
}

export default Autocomplete;
