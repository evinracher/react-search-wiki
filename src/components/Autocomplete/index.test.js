import React from 'react';
import Autocomplete from '.';
import { shallow } from 'enzyme';

const render = (props) => shallow(<Autocomplete {...props} />);

describe('Autocomplete component', () => {
  let sut;
  let props;

  describe('without props', () => {
    beforeEach(() => {
      sut = render(props);
    });

    it('should match snapshot', () => {
      expect(sut).toMatchSnapshot();
    });
  });

  describe('when articles are passed', () => {
    beforeEach(() => {
      props = {
        articles: [],
        searchValue: 'search value',
        onSearchChange: jest.fn()
      };

      sut = render(props);
    });

    it('should match renderMenu snapshot', () => {
      const props = sut.props();

      expect(props.renderMenu()).toMatchSnapshot();
    });
  });

  describe('when articles are not passed', () => {
    beforeEach(() => {
      props = {
        articles: null,
        searchValue: 'search value',
        onSearchChange: jest.fn()
      };

      sut = render(props);
    });

    it('should match renderMenu snapshot', () => {
      const props = sut.props();

      expect(props.renderMenu()).toMatchSnapshot();
    });
  });

  describe('when renderItem props is called', () => {
    beforeEach(() => {

      sut = render(props);
    });

    describe('when item is highlighted', () => {
      it('should match renderItem snapshot', () => {
        const { renderItem } = sut.props();
        const item = {
          id: 1,
          label: 'Test'
        };

        expect(renderItem(item, true)).toMatchSnapshot();
      });
    });

    describe('when item is not highlighted', () => {
      it('should match renderItem snapshot', () => {
        const { renderItem } = sut.props();
        const item = {
          id: 1,
          label: 'Test'
        };

        expect(renderItem(item, false)).toMatchSnapshot();
      });
    });

  });

  describe('when getItemValue is called', () => {
    beforeEach(() => {
      sut = render(props);
    });

    it('should return label', () => {
      const { getItemValue } = sut.props();

      expect(getItemValue({label: 'test'})).toBe('test');
    });
  });
});