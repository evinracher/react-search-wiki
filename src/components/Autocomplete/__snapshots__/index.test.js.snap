// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete component when articles are not passed should match renderMenu snapshot 1`] = `<React.Fragment />`;

exports[`Autocomplete component when articles are passed should match renderMenu snapshot 1`] = `<React.Fragment />`;

exports[`Autocomplete component when renderItem props is called when item is highlighted should match renderItem snapshot 1`] = `
<div
  className="input-suggestions-item"
  style={
    Object {
      "backgroundColor": "#eee",
    }
  }
>
  <a
    className="input-suggestions-link"
    href={1}
    target="_blank"
  >
    Test
  </a>
</div>
`;

exports[`Autocomplete component when renderItem props is called when item is not highlighted should match renderItem snapshot 1`] = `
<div
  className="input-suggestions-item"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <a
    className="input-suggestions-link"
    href={1}
    target="_blank"
  >
    Test
  </a>
</div>
`;

exports[`Autocomplete component without props should match snapshot 1`] = `
<Autocomplete
  autoHighlight={true}
  getItemValue={[Function]}
  inputProps={
    Object {
      "placeholder": "Input a search term",
    }
  }
  isItemSelectable={[Function]}
  menuStyle={
    Object {
      "background": "rgba(255, 255, 255, 0.9)",
      "borderRadius": "3px",
      "boxShadow": "0 2px 12px rgba(0, 0, 0, 0.1)",
      "fontSize": "90%",
      "maxHeight": "50%",
      "overflow": "auto",
      "padding": "2px 0",
      "position": "fixed",
    }
  }
  onChange={[Function]}
  onMenuVisibilityChange={[Function]}
  onSelect={[Function]}
  renderInput={[Function]}
  renderItem={[Function]}
  renderMenu={[Function]}
  selectOnBlur={false}
  value=""
  wrapperProps={Object {}}
  wrapperStyle={
    Object {
      "display": "inline-block",
    }
  }
/>
`;
